File: 0392-is-subsequence/0392-is-subsequence.dart
class Solution {
  bool isSubsequence(String s, String t) {
    int len=0;
    if(s.isEmpty)
        return true;
        if(s.length>t.length)
            return false;
    for(int i=0;i<t.length;i++){
        if(s[len]==t[i])
            len++;
        if(len==s.length)
            return true;
    }
    return false;
  }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
          int i=0,j=0;
        int ctr=0;
        while(i<t.size()&&j<s.size()){
            if(t[i]==s[j]){
                ctr++;
                j++;
            }
            i++;
        }
        if(ctr==s.size()){
            return true;
        }
        return false;
    }
};